include/cppoptlib/linesearch/armijo.h

```diff
@@ -24,6 +24,17 @@ class Armijo {
         const Scalar c = 0.2;
         const Scalar rho = 0.9;
         Scalar alpha = alpha_init;
+        int iter;
+        for (iter = 0; iter < 200; iter++) {
```

Comment:
We should check if these improvements speedup the benchmark scripts. I am ok with the 200, as it turns out all commercial solver just do hard-core parameter tuning.

One issue might be that the solver walks tiny steps along the constraints. I am not sure if this clipping should be be done in the line search or solver.